lib/version-fetcher.js

```diff
@@ -1,44 +1,54 @@
-const fs = require('fs');
+const fs = require('fs-extra');
 const tmp = require('tmp');
 const path = require('path');
-const {execSync} = require('child_process');
+const execa = require('execa');
 const packageHandler = require('./package-handler');
 
-function copyNpmrc(dirName, cwd) {
+function tmpDirAsync(opts) {
+  return new Promise((res, rej) =>
+    tmp.dir(opts, (err, name) => err ? rej(err) : res(name)));
```

Comment:
I think the `tmp.dir` signature can be used with `require('util').promisify` if you want to make it shorter